spark3/src/test/java/org/apache/iceberg/spark/source/TestStructuredStreamingRead3.java

```diff
@@ -165,6 +165,84 @@ public void testReadStreamOnIcebergTableWithMultipleSnapshots() throws Exception
     Assertions.assertThat(actual).containsExactlyInAnyOrderElementsOf(Iterables.concat(expected));
   }
 
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testReadingStreamFromTimestamp() throws Exception {
+    List<SimpleRecord> dataBeforeTimestamp = Lists.newArrayList(
+        new SimpleRecord(-2, "minustwo"),
+        new SimpleRecord(-1, "minusone"),
+        new SimpleRecord(0, "zero"));
+    appendData(dataBeforeTimestamp, tableIdentifier, "parquet");
+
+    table.refresh();
+    long streamStartTimestamp = table.currentSnapshot().timestampMillis() + 1;
```

Comment:
This needs to use `waitUntilAfter`.

Reply:
+1. Under load in CI, time based tests can get flakey and waitUntilAfter will help guard against that.

Reply:
I have add the waitUntilAfter to make sure current system time is passed the expected snapshot time

Reply:
am resolving the comments - which are addressed - to let the ones that needs resolution pop-out. Pl. definitely re-activate the comment - if you feel the need @kbendick / @rdblue.